File: src/main/java/net/cr24/primeval/util/ShapesUtil.java
package net.cr24.primeval.util;

import java.util.ArrayList;

public class ShapesUtil {

    public static IntPoint2D[] getCoordinatesInCircle(int radius) {
        ArrayList<IntPoint2D> points = new ArrayList<>(radius);
        int radiusSquared = radius * radius;
        for (int x = -radius; x <= radius; x++) {
            for (int y = -radius; y <= radius; y++) {
                if (x*x + y*y <= radiusSquared) {
                    points.add(new IntPoint2D(x, y));
                }
            }
        }
        return points.toArray(new IntPoint2D[points.size()]);
    }

}

File: src/main/java/net/cr24/primeval/util/ShapesUtil.kt
package net.cr24.primeval.util

class ShapesUtil {
    companion object {
        fun getCoordinatesInCircle(radius: Int): Array<IntPoint2D> {
            val points = ArrayList<IntPoint2D>(radius)
            val radiusSquared = radius * radius
            for (x in -radius..radius) {
                for (y in -radius..radius) {
                    if (x * x + y * y <= radiusSquared) {
                        points.add(IntPoint2D(x, y))
                    }
                }
            }
            return points.toTypedArray()
        }
    }
}